Bài 9. Tập dữ liệu sau cho ta điểm thi của một số sinh viên (thang điểm 100). (Dữ liệu có trong file DiemThi.xls)

a. Hãy tính các số đo hướng tâm của tập dữ liệu: trung bình cộng, trung vị và
mode. Nêu ý nghĩa của những giá trị này.

In [ ]:
import pandas as pd

In [ ]:
dthi = pd.read_excel('DiemThi.xlsx') # phải save as file DiemThi.xls sang file xlsx
print(dthi.head()) # in ra 5 dòng đầu tiên

   DiemThi
0       61
1       27
2       26
3       37
4       30


#### 1. Tính trung bình mẫu

In [ ]:
dthi['DiemThi'].mean() # Tính mean (trung bình cộng) của điểm thi
# Trung bình cộng: 53.3
#YN: điểm thi trung bình của các thí sinh là 53.3

53.3

In [ ]:
# Hoặc dùng numpy
import numpy as np
np.mean(dthi.DiemThi)

53.3

#### 2. Tính trung vị

In [ ]:
dthi['DiemThi'].median() # tính trung vị của điểm thi
# Trung vị là 48
# YN: Khoảng 50% số thí sinh có điểm thi <= 48
# và khoảng 50% số thí sinh có điểm thi >=48

48.0

#### 3. Tìm mode (là giá trị có tần số lớn nhất)

In [ ]:
# Cách 1:
dthi['DiemThi'].value_counts() # bảng tần số, từ đó tìm được mode là điểm có tần số lớn nhất
# Vậy có 2 mode là 60 và 47
# YN: Có nhiều thí sinh được 60 điểm và 47 điểm nhất

60     4
47     4
48     3
61     2
37     2
30     2
90     2
25     2
45     2
57     1
95     1
80     1
28     1
35     1
53     1
65     1
100    1
27     1
81     1
19     1
67     1
46     1
63     1
87     1
26     1
55     1
Name: DiemThi, dtype: int64

In [ ]:
# Cách 2: kết quả ra tương tự nhưng output gọn hơn
tsdt = dthi.DiemThi.value_counts() # bảng tần số của điểm thi
tsdt[tsdt==max(tsdt)] # tìm giá trị có tần số lớn nhất trong bảng tần số

60    4
47    4
Name: DiemThi, dtype: int64

b. Tính các đại lượng mô tả độ phân bố của tập dữ liệu như: tứ phân vị, phân vị
thứ 10,60,90 của tập dữ liệu. Nêu ý nghĩa của những giá trị này.

#### 4. Tìm tứ phân vị

In [ ]:
dthi['DiemThi'].quantile([0.25, 0.5, 0.75])   #Tính các tứ phân vị

0.25    37.0
0.50    48.0
0.75    61.5
Name: DiemThi, dtype: float64

Tứ phân vị thứ nhất là 37 => ý nghĩa: Khoảng 25% số thí sinh có điểm thi <= 37  và khoảng 75% số thí sinh có điểm thi >= 37

Tứ phân vị thứ 2 là 48 (chính là trung vị)

Tứ phân vị thứ 3 là 61.5 => ý nghĩa: Khoảng 75% số thí sinh có điểm thi <= 61.5  và khoảng 25% số thí sinh có điểm thi >= 61.5

#### 5. Tìm phân vị bất kì

In [ ]:
dthi['DiemThi'].quantile([0.1, 0.6, 0.9]) # tính phân vị thứ 10, 60 và 90

0.1    26.9
0.6    58.2
0.9    87.3
Name: DiemThi, dtype: float64

Phân vị thứ 10 là 26.9 => ý nghĩa: Khoảng 10% số thí sinh có điểm thi <= 26.9  và khoảng 90% số thí sinh có điểm thi >= 26.9

Phân vị thứ 60 là 58.2, ý nghĩa tương tự

Phân vị thứ 90 là 87.3, ý nghĩa tương tự

c. Tính các đại lượng mô tả độ phân tán của tập dữ liệu: khoảng biến thiên, độ trải giữa, phương sai và độ lệch chuẩn.

#### 6. Tìm khoảng biến thiên

In [ ]:
max(dthi.DiemThi)-min(dthi.DiemThi) # tính khoảng biến thiên

81

Khoảng biến thiên là 81

#### 7. Tìm phương sai

In [ ]:
dthi['DiemThi'].var() # tính phương sai

441.1384615384614

Phương sai là 441.14

#### 8. Tìm độ lệch chuẩn

In [ ]:
dthi['DiemThi'].std() # tính độ lệch chuẩn

21.003296444569393

Độ lệch chuẩn là 21

9. Tìm độ trải giữa: Độ trải giữa = Q3 - Q1, trong đó Q3, Q1 lần lượt là tứ phân vị thứ 3 và thứ nhất.

In [2]:
61.5-37 # Tìm độ trải giữa

24.5